sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -211,7 +211,7 @@ class Dataset[T] private[sql](
 
   private implicit def classTag = unresolvedTEncoder.clsTag
 
-  def sqlContext: SQLContext = sparkSession.wrapped
+  lazy val sqlContext: SQLContext = sparkSession.wrapped
```

Comment:
also please add a comment explaining that it must be a val because stable identifier is expected when you import the implicits